esy/Task.ml

```diff
@@ -119,6 +120,7 @@ let buildId
       Package.CommandList.show pkg.buildCommands;
       Package.CommandList.show pkg.installCommands;
       Package.BuildType.show pkg.buildType;
+      Package.SandboxEnv.show rootPkg.sandboxEnv;
```

Comment:
@andreypopp Even if we don't set any sandbox env vars, esy will still reset all the build ids because of the way hash is currently calculated: we're joining all the pieces with `--`, so empty sandbox env will result in calculating the hash from some `name--version--buildcmd--[]--otherstuff`.

Should we make sandbox env string part of something else? Like:

`pkg.version ^ (Package.SandboxEnv.showIfNotEmpty rootPkg.sandboxEnv);`